redfish-core/lib/systems.hpp

```diff
@@ -51,6 +55,56 @@
 namespace redfish
 {
 
+/* @brief validate systemName and get system index from systemName
+ *
+ * @param[in] systemName     requested system name
+ *
+ * @return int    -1 on failure, else validatet index
+ * */
+inline int getSystemIndex(const std::string_view systemName)
+{
+#ifndef BMCWEB_ENABLE_MULTI_COMPUTERSYSTEM
+    // on single host we only want system0 / system to be valid
+    if (systemName == "system0" || systemName == "system")
+    {
+        return 0;
+    }
+    return -1;
+#else
+
+    static constexpr std::string_view systemToken = "system";
+    auto location = systemName.rfind(systemToken);
+
+    if (location == std::string::npos)
+    {
+        BMCWEB_LOG_DEBUG("invalid host number");
+        return -1;
+    }
+
+    // TODO  12/05/23-14:37 olekatpyle: only works for single digit number
+    // of hosts, need to add support for multi digit host numbers
+    auto index = systemName.substr(location + systemToken.length());
+    if (index.empty())
+    {
+        return 0;
+    }
+
+    // TODO  12/05/23-14:36 olekatpyle: also validate range of index against
+    // number of available hosts, so that system4 is not valid if only 3 hosts
+    // exist?
+
+    auto hostNumber = std::stoi(std::string(index));
+
+    if (std::in_range<size_t>(hostNumber))
```

Comment:
this should be 
```
std::in_range<int>(hostNumber);
```
since the function returns int. And also should check >= 0. Additionally, std::stoi can throw, unsure if it should be catched here.

https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
Ack